### ISSUE TITLE
An empty string in the result when `code_to_run` contains built-ins which is not in the whitelist

### ISSUE THREAD

**Issue by nautics889** [gventuri/pandas-ai#363]
### 🐛 Describe the bug

**Context:**

Interpreter: 3.11.4 (tags/v3.11.4:d2340ef, Jun  7 2023, 05:45:37) [MSC v.1934 64 bit (AMD64)]
pandas-ai version: HEAD from the main branch (0.6.7)
___
**Description:**

When investigating issue #353 i slightly modified the code from there, here is how it's look like:

```
import pandas as pd
from pandasai import PandasAI

# Sample DataFrame
df = pd.DataFrame({
    "house": ["a", "b", "c"],
    "happiness_index": [6.94, 7.16, 6.66],
	"people": [['abby', 'ben'], ['ken'], ['ian']]
})

# Instantiate a LLM
from pandasai.llm.openai import OpenAI
llm = OpenAI()

pandas_ai = PandasAI(llm)
pandas_ai.clear_cache()

response = pandas_ai.run(df, prompt='Which house has at least two people?')
print(response)  # empty string

```

While inspecting the behaviour of `PandasAI`, i found that [the `code_to_run` variable](https://github.com/gventuri/pandas-ai/blob/main/pandasai/__init__.py#L647) contains the `eval()` function from built-ins:
```
In [2]: code_to_run
Out[2]: "df[df['people'].apply(lambda x: len(eval(x))) >= 2]['house']"
```
Meanwhile we don't have `eval()` in the environment:
```
In [3]: 'eval' in environment['__builtins__']
Out[3]: False
```
...as well as in [`WHITELISTED_BUILTINS`](https://github.com/gventuri/pandas-ai/blob/main/pandasai/constants.py#L2).
This caused an Exception being raised from [here](https://github.com/gventuri/pandas-ai/blob/main/pandasai/__init__.py#L675).

___
**Solution**
For now i'd propose just to add logging [somewhere in exception catching](https://github.com/gventuri/pandas-ai/blob/main/pandasai/__init__.py#L675), or check that `code_to_run` contains some built-ins that are not white-listed.

**Comment by gventuri** [comment]
@nautics889 totally agree. I would probably go for the option 1, as option 2 seems difficult (in fact we whitelist, so it's hard to parse in search for blacklisted builtins). What do you think?

**Comment by nautics889** [comment]
@gventuri thank you for the reply!
I fully agree as well, an additional log message for this would be enough as seems to me.
I'd appreciate if you could take a look at [PR #371](https://github.com/gventuri/pandas-ai/pull/371) when you have time. 
Log levels and message string which i've added was taken on my own, so it's surely open to discuss — please let me know, if something needs to be changed.
